app/models/concerns/user/low_balance_fraud_check.rb

```diff
@@ -29,12 +32,67 @@ def check_for_low_balance_and_probate(refunded_or_disputed_purchase_id)
     disable_refunds_and_put_on_probation! unless recently_probated_for_low_balance?
   end
 
+  def check_for_balance_recovery_and_mark_compliant
+    return unless on_probation?
+    return unless probated_by_low_balance_fraud_check?
+    return if unpaid_balance_cents.nil? || unpaid_balance_cents <= BALANCE_RECOVERY_THRESHOLD_IN_CENTS
+
+    previous_state = get_previous_risk_state_before_probation || "compliant"
+
+    content = "Automatically restored to #{previous_state} state on #{Time.current.to_fs(:formatted_date_full_month)} as balance recovered above $100"
+
+    restore_to_previous_state!(previous_state, author_name: LOW_BALANCE_FRAUD_CHECK_AUTHOR_NAME, content: content)
+  end
+
   private
+    def get_previous_risk_state_before_probation
+      probation_version = versions
+        .where("object_changes LIKE ?", "%user_risk_state%")
+        .order(created_at: :desc)
+        .find do |version|
+          changes = PaperTrail.serializer.load(version.object_changes)
+          changes["user_risk_state"]&.last == "on_probation"
+        end
+
+      return nil unless probation_version
+
+      changes = PaperTrail.serializer.load(probation_version.object_changes)
+      changes["user_risk_state"]&.first
+    end
+
+    def restore_to_previous_state!(state, author_name:, content:)
+      case state
+      when "not_reviewed"
+        mark_not_reviewed!(author_name: author_name, content: content)
+      when "compliant"
+        mark_compliant!(author_name: author_name, content: content)
+      when "flagged_for_fraud"
+        flag_for_fraud!(author_name: author_name, content: content)
+      when "flagged_for_tos_violation"
+        flag_for_tos_violation!(author_name: author_name, content: content)
+      else
+        mark_compliant!(author_name: author_name, content: content)
+      end
+    end
+
+    def probated_by_low_balance_fraud_check?
+      comments.with_type_on_probation
+              .where(author_name: LOW_BALANCE_FRAUD_CHECK_AUTHOR_NAME)
+              .exists?
```

Comment:
This is insufficient.
- If the comment predates the a transition to a compliant state, this should return false.
- If a probation comment done manually by an admin is more recent, this should return false. This automatic restoring behavior only applies to user that have _automatically_ been put on probation by the system.

This needs to lookup the most recent probation comment and check whether it was done by the system